pkg/apis/monitoring/v1/types.go

```diff
@@ -607,6 +607,10 @@ func (o *OAuth2) Validate() error {
 		}
 	}
 
+	if o.TLSConfig == nil {
```

Comment:
Should we move the test to `*SafeTLSConfig.Validate()`?

Reply:
move nil check to *SafeTLSConfig.Validate()